services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/validation/Validators.kt

```diff
@@ -98,6 +98,25 @@ sealed class ValidationResult {
         return this
     }
 
+    /**
+     * If either this or the validated function is ok, return ok.
+     *
+     * If both have errors, return all errors.
```

Comment:
"return ok" and "return all errors" sound a bit misleading since the returned object is always `this` ValidationResult. Maybe "return without errors" and "return with all errors"?

Reply:
The returned object can also be the result of the given validation function, when `this` has errors, but the new result doesn't.

I made ValidationResult immutable. Now the comments should be correct?